diff --git a/node_modules/react-native-webview/apple/RNCWebViewImpl.m b/node_modules/react-native-webview/apple/RNCWebViewImpl.m
index ed3a81a..7d361e7 100644
--- a/node_modules/react-native-webview/apple/RNCWebViewImpl.m
+++ b/node_modules/react-native-webview/apple/RNCWebViewImpl.m
@@ -474,9 +474,12 @@ - (WKWebViewConfiguration *)setUpWkWebViewConfig
   return wkWebViewConfig;
 }
 
-- (void)didMoveToWindow
+- (void)initializeWebView
 {
-  if (self.window != nil && _webView == nil) {
+   @synchronized (self) {
+    if (_webView != nil) {
+      return;
+    }
     WKWebViewConfiguration *wkWebViewConfig = [self setUpWkWebViewConfig];
     _webView = [[RNCWKWebView alloc] initWithFrame:self.bounds configuration: wkWebViewConfig];
     [self setBackgroundColor: _savedBackgroundColor];
@@ -529,6 +532,23 @@ - (void)didMoveToWindow
     [self setKeyboardDisplayRequiresUserAction: _savedKeyboardDisplayRequiresUserAction];
     [self visitSource];
   }
+}
+
+// react-native-mac os does not support didMoveToSuperView
+#if !TARGET_OS_OSX
+- (void)didMoveToSuperview
+{
+  if (self.superview != nil && _webView == nil) {
+    [self initializeWebView];
+  }
+}
+#endif // !TARGET_OS_OSX
+
+- (void)didMoveToWindow
+{
+  if (self.window != nil && _webView == nil) {
+    [self initializeWebView];
+  }
 #if !TARGET_OS_OSX
   // Allow this object to recognize gestures
   if (self.menuItems != nil) {
